Userland/Libraries/LibELF/DynamicObject.h

```diff
@@ -162,6 +162,25 @@ class DynamicObject : public RefCounted<DynamicObject> {
         GNU
     };
 
+    class HashSymbol {
+    public:
+        HashSymbol(const StringView& name)
+            : m_name(name)
+        {
+        }
+
+        StringView name() const { return m_name; }
+        u32 gnu_hash() const;
+        u32 sysv_hash() const;
+
+    private:
+        StringView m_name;
+        mutable u32 m_gnu_hash;
+        mutable bool m_gnu_hash_used { false };
+        mutable u32 m_sysv_hash;
+        mutable bool m_sysv_hash_used { false };
```

Comment:
Seems like `mutable Optional<u32>` would be a slightly more idiomatic fit here.

Reply:
Aye, good point. Fixed. :)